sanctum-solana-client-utils/src/prioritization_fee.rs

```diff
@@ -0,0 +1,152 @@
+use std::cmp::min;
+
+use medians::Medianf64;
+use solana_client::{
+    nonblocking::rpc_client::RpcClient as NonblockingRpcClient,
+    rpc_client::{RpcClient, SerializableTransaction},
+    rpc_response::RpcPrioritizationFee,
+};
+use solana_rpc_client_api::{
+    client_error::Error as ClientError, config::RpcSimulateTransactionConfig,
+};
+use solana_sdk::{
+    compute_budget::ComputeBudgetInstruction, instruction::Instruction, pubkey::Pubkey,
+};
+
+const WEIGHTED_MEDIAN_EPSILON: f64 = 0.0001;
+
+/// Calculate slot weighted median value of given sample of prioritization fees
+/// (see get_recent_prioritization_fees rpc call)
+///
+/// Weights are assigned such that the values would range from 1/(# of slots) to 1
+///
+/// NOTE: assumes unreported slots has `prioritization_fee = 0`
+pub fn calc_slot_weighted_median_prioritization_fees(
+    rpc_prio_fees: &[RpcPrioritizationFee],
+) -> Option<u64> {
+    let max_slot = rpc_prio_fees.iter().max_by_key(|fee| fee.slot)?.slot;
+    let min_slot = rpc_prio_fees.iter().min_by_key(|fee| fee.slot)?.slot;
+    let slot_interval = max_slot - min_slot + 1;
+
+    let (values, weights): (Vec<f64>, Vec<f64>) = rpc_prio_fees
+        .iter()
+        .filter_map(|fee| {
+            if fee.prioritization_fee == 0 {
+                None
+            } else {
+                Some((
+                    fee.prioritization_fee as f64,
+                    (1 + fee.slot - min_slot) as f64 / slot_interval as f64,
```

Comment:
this line crashed sanctum-s-api server with `panicked at 'attempt to subtract with overflow'` on rev [867be34](https://github.com/igneous-labs/sanctum-solana-utils/pull/45/commits/867be34eebff6acd1929290d20acbbd33e5bb11f) so its good that we changed the min_slot definition to be the min of the sample here